### ISSUE TITLE
Remove semaphore from Release

### ISSUE THREAD

**Issue by squaremo** [weaveworks/flux#258]
I don't think there's any point to it, given that the model is surely to have a pool of workers that run through jobs one each at a time, and no need for mutual exclusion except on release jobs themselves.

**Comment by squaremo** [comment]
They* are running on a ticker as well, for some reason. They should run like this:

```
Is there a job to do?
  Yes: claim it
  No: wait for a bit, then go to the start
```

*There's actually only one, but there's no reason why there couldn't be more.

**Comment by squaremo** [comment]
Fixed by #296
